scipy/sparse/_lil.py

```diff
@@ -287,6 +287,9 @@ def _mul_scalar(self, other):
     def __truediv__(self, other):           # self / other
         if isscalarlike(other):
             new = self.copy()
+            new_dtype = np.result_type(self.dtype, np.array(other).dtype)
+            if new_dtype != self.dtype:
+                new.data = new.data.astype(new_dtype)
```

Comment:
This won't work. The `data` member of LIL-format objects is a numpy array of Python lists, with dtype=object.

To change the dtype, we need to do something like this:

```
for i, rowvals in enumerate(new.data):
    new.data[i] = rowvals.astype(new_dtype)
```

Reply:
Sorry, I should have noticed this. Isn't just the following what we really want?
```python
new.dtype =  np.result_type(self, other)
```
The list data is actually modified correctly. Things only start breaking when trying to convert to a different storage format, for example.